<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CameraWidget</class>
 <widget class="QWidget" name="CameraWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>580</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Camera</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>578</width>
        <height>380</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="contentLayout">
       <item>
        <widget class="QGroupBox" name="camera_basic_group">
         <property name="title">
          <string>Cấu hình cơ bản camera</string>
         </property>
         <layout class="QFormLayout" name="basic_layout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_camera_index">
            <property name="text">
             <string>Chỉ số camera:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="camera_index_layout">
            <item>
             <widget class="QSpinBox" name="camera_index_spin">
              <property name="minimum">
               <number>0</number>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="scan_camera_btn">
              <property name="text">
               <string>Quét camera</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_frame_width">
            <property name="text">
             <string>Chiều rộng khung hình:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="frame_width_spin">
            <property name="minimum">
             <number>320</number>
            </property>
            <property name="maximum">
             <number>3840</number>
            </property>
            <property name="singleStep">
             <number>160</number>
            </property>
            <property name="value">
             <number>640</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_frame_height">
            <property name="text">
             <string>Chiều cao khung hình:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="frame_height_spin">
            <property name="minimum">
             <number>240</number>
            </property>
            <property name="maximum">
             <number>2160</number>
            </property>
            <property name="singleStep">
             <number>120</number>
            </property>
            <property name="value">
             <number>480</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_fps">
            <property name="text">
             <string>Tốc độ khung hình (FPS):</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="fps_spin">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>60</number>
            </property>
            <property name="value">
             <number>30</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="vl_group">
         <property name="title">
          <string>Cấu hình mô hình ngôn ngữ hình ảnh</string>
         </property>
         <layout class="QFormLayout" name="vl_layout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_local_vl_url">
            <property name="text">
             <string>URL VL cục bộ:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="local_vl_url_edit">
            <property name="placeholderText">
             <string>Ví dụ: https://open.bigmodel.cn/api/paas/v4/</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_vl_api_key">
            <property name="text">
             <string>Khóa API VL:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="vl_api_key_edit">
            <property name="echoMode">
             <enum>QLineEdit::Password</enum>
            </property>
            <property name="placeholderText">
             <string>Vui lòng nhập khóa API của bạn</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_models">
            <property name="text">
             <string>Tên mô hình:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="models_edit">
            <property name="placeholderText">
             <string>Ví dụ: glm-4v-plus</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="preview_group">
         <property name="title">
          <string>Xem trước camera</string>
         </property>
         <layout class="QVBoxLayout" name="preview_layout">
          <item>
           <widget class="QLabel" name="preview_label">
            <property name="minimumSize">
             <size>
              <width>320</width>
              <height>240</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>480</width>
              <height>360</height>
             </size>
            </property>
            <property name="styleSheet">
             <string>border: 2px solid #ccc; background-color: #000;</string>
            </property>
            <property name="text">
             <string>Khu vực xem trước camera
Nhấn để làm mới xem trước hoặc thay đổi chỉ số camera</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="preview_controls_layout">
            <item>
             <widget class="QPushButton" name="start_preview_btn">
              <property name="text">
               <string>Bắt đầu xem trước</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="stop_preview_btn">
              <property name="text">
               <string>Dừng xem trước</string>
              </property>
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="preview_spacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
